### DISCUSSION TITLE
Calibration error

### DISCUSSION THREAD

**Discussion by Emina9595** [iiasa/CWatM#172] [Q&A]
Hello!

I was trying to run calibration for 30-arcminute resolution CWatM model, but after running 
_**1a. run_single_calibration.bat**_ I keep receiving this message which I don't know how to interpet. 
![Calibration_error](https://github.com/user-attachments/assets/eb5909c9-808f-43e8-9722-138ab9fcdbb7)

Thanks in advance for any suggestion!

Emina

**Comment by mikhailsmilovic** [comment]
Guten Tag @Emina9595 Emina, would you switch to the develop branch? 
This has been fixed but has yet to be integrated into the main CWatM version.

Switching may overwrite some of the files (ex., settings file) you have updated. 
To avoid losing your changes, copy these files somewhere before switching branches.

Cheers!

**Reply by Emina9595** [reply]
Thank you Mikhail!

I have changed main branch to develop. Now there is another error. 

![calibration error_2](https://github.com/user-attachments/assets/953cf9d3-1875-4bd6-b994-be06722e8392)

And here is the setting file for calibration:
[settings_calibration.txt](https://github.com/user-attachments/files/17994938/settings_calibration.txt)

Best regards!

**Reply by mikhailsmilovic** [reply]
Guten Tag, would you share your observations file (Sava_discharge_2001_2002.csv)?

**Comment by Emina9595** [comment]
[Sava_discharge_2001_2002.csv](https://github.com/user-attachments/files/17995790/Sava_discharge_2001_2002.csv)

Of course, here it is.

**Reply by mikhailsmilovic** [reply]
Let us experiment ⚗️
Would you remove the two commas at the end of each line and try again?

**Reply by Emina9595** [reply]
Do you mean, rounding numbers?
![image](https://github.com/user-attachments/assets/9a094c0d-94a7-41e0-b0d5-18a000f55bf9)

**Reply by Emina9595** [reply]
I did that anyway, and now got another error :( 
![image](https://github.com/user-attachments/assets/705a4a15-3dee-4919-9c54-628f6d20f0b8)

...but at least, these folders appeared for the first time :) 

![image](https://github.com/user-attachments/assets/8b37275e-695e-4dcd-b6c8-cbd231d00cfa)

**Reply by mikhailsmilovic** [reply]
You can go into one of the folders, and there will be a log file. 
This should give you a hint as to why the model didn't run.

**Reply by Emina9595** [reply]
Thank you Mikhail! The log files indicate that I need to downgrade NumPy. 
![image](https://github.com/user-attachments/assets/ffab3b23-c44c-46e7-b4aa-41b645c0d562)
I also found the requirements in the discussions:
![image](https://github.com/user-attachments/assets/42c8f48d-4bee-4071-8399-815076822e15)
So I tried to downgrade NumPy using "pip install --upgrade numpy==1.23.4" command, but I think unsuccessfully 
![image](https://github.com/user-attachments/assets/d3ecba25-0101-4074-adcf-f6cab85c6a4b)
![image](https://github.com/user-attachments/assets/1bb4e19c-f42b-43aa-ac52-85c8b19ae1a4)

I feel like I'm stuck now with this error, and really don't know how to solve it...

Also when I try to install Numpy 1.23.4 within my working environment (cwatm) I get this message:
![image](https://github.com/user-attachments/assets/933c8b22-550a-4d03-bb85-ed3f99090086)

**Reply by Emina9595** [reply]
Also, when I tried to fetch changes, this note appeared:
![image](https://github.com/user-attachments/assets/fefb131d-25f1-4019-9b26-a004322f215f)
Not sure if this has to do with me trying to install Numpy, or something else...

**Comment by mikhailsmilovic** [comment]
Guten Tag @Emina9595 
- Given the trouble with the packages, I could recommend creating a new virtual environment and re-installing the packages, starting right away with the earlier Numpy. It's a little work, but it will solve this.
- To accept the update from GitHub, one must first discard their changes (highlight the files, right click and select discard changes). To save your changes in any settings file, first copy and save them elsewhere, and bring them back later.

Let me know how it goes!

**Reply by Emina9595** [reply]
Guten Morgen @mikhailsmilovic 

Thank you ! I will let you know how it went :)

**Reply by Emina9595** [reply]
Guten Abend @mikhailsmilovic :)

I reinstalled python and everything to my laptop (started completely fresh) so now the calibration is running!
Although there are few notes, but I don't think those are errors...
![image](https://github.com/user-attachments/assets/3bb333f1-3ca4-4e47-9d4f-5c29d018baee)
Right now it came up to the 2nd generation I think... Hope it will end well! Thank you for the support :)
